proknow/Audit.py

```diff
@@ -0,0 +1,159 @@
+
+        if "types" in kwargs: 
+            if not isinstance(kwargs["types"], list):
+                self.options["types"] = [kwargs["types"]]
+            else:
+                self.options["types"] = kwargs["types"]
+
+        if "methods" in kwargs: 
+            if not isinstance(kwargs["methods"], list):
+                self.options["methods"] = [kwargs["methods"].upper()]
+            else:
+                self.options["methods"] = [x.upper() for x in kwargs["methods"]]
+
+        if "status_codes" in kwargs: 
+            if not isinstance(kwargs["status_codes"], list):
+                self.options["status_codes"] = [kwargs["status_codes"]]
+            else:
+                self.options["status_codes"] = kwargs["status_codes"]
+
+        if "text" in kwargs: 
+            if not isinstance(kwargs["text"], list):
+                self.options["text"] = [kwargs["text"]]
+            else:
+                self.options["text"] = kwargs["text"]
+
+        self.auditResultsPage = self._query()
+
+        self.options['page_number'] = 0
+        if self.auditResultsPage.total > 0:
+            self.options['first_id'] = self.auditResultsPage.items[0]['id']
+
+        return self.auditResultsPage
+
+    def next(self):
+        """Gets the next page of results using the parameters provided by the 
+        :meth:`proknow.Audit.Audit.query` method.
+        
+        Returns:
+            :class:`proknow.Audit.AuditResultsPage`
+
+        Raises:
+            :class:`proknow.Exceptions.HttpError`: If the HTTP request generated an error.
+            :class:`proknow.Exceptions.InvalidOperationError`: If this method is called before :meth:`proknow.Audit.Audit.query` is called.
+        """
+        if self.auditResultsPage is None:
+            raise InvalidOperationError('\'Audit.query\' must be called before calling \'Audit.next\'')
+
+        self.options['page_number'] += 1
+        self.auditResultsPage = self._query()
+
+        return self.auditResultsPage
+
+    def _query(self): 
+        _, results = self._requestor.post('/audit/events/search', json=self.options)
+        
+        return AuditResultsPage(self, results['total'], results['items'])
+
+class AuditResultsPage(object):
+    """
+    This class represents a page of audit log query results. It's instantiated by the
+    :meth:`proknow.Audits.Audit.query` method.
+
+    Attributes:
+        total (int): The total number of possible results for the given query, not the total of items in the object's item list.
+        items (list of dict): A list of dictionaries to represent each item returned for the current page.
+    """
+    def __init__(self, audit, total, items):
+        self._audit = audit
+        self.total = total
+        self.items = items
+    
+    def next(self):
+        """Gets the next page of query results using the initial query parameters.
+
+        Returns:
+            :class:`proknow.Audit.AuditResultsPage`
```

Comment:
See previous note about the return type.

Reply:
Take a look at this one again. Please add a description to the return type here.